nixos/modules/services/web-apps/node-red.nix

```diff
@@ -0,0 +1,79 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.node-red;
+in
+{
+  options.services.node-red = {
+    enable = mkEnableOption "the Node-RED service";
+
+    package = mkOption {
+      default = pkgs.nodePackages.node-red;
+      defaultText = "pkgs.nodePackages.node-red";
+      type = types.package;
+      description = "Node-RED package to use.";
+    };
+
+    configFile = mkOption {
+      type = types.path;
+      default = "${cfg.package}/lib/node_modules/node-red/settings.js";
+      defaultText = "\${cfg.package}/lib/node_modules/node-red/settings.js";
+      description = ''
+        Path to the JavaScript configuration file.
+        See <link
+        xlink:href="https://github.com/node-red/node-red/blob/master/packages/node_modules/node-red/settings.js"/>
+        for a configuration example.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 1880;
+      description = "Listening port.";
+    };
+
+    userDir = mkOption {
```

Comment:
It doesn't seem like this is a valuable option. It seems like it will cause more problem than anything. What is going to happen if the sysadmin changes this value? In almost every scenario the application will likely fail to start, or start in a degraded state. I would suggest removing this.

Reply:
> What is going to happen if the sysadmin changes this value?

What do you mean by this? Can you give me an example of sysadmins changing this value?
 
Most services in NixOS have the option to declare where the state is stored, why should this be any different?

Reply:
Have I corrected this with this commit @aanderse?

cd7284d

Reply:
`userDir` still isn't created if the user specifies anything other than the default, and `/var/lib/node-red` is created even if the user specifies something else. How about [this](https://github.com/NixOS/nixpkgs/commit/42219a0df086a580d96e54dbd65d6f081c6dc20a)?